### ISSUE TITLE
"Sort by Imported" is splitting an ablum

### ISSUE THREAD

**Issue by liu3631** [Taiko2k/TauonMusicBox#635]
I've recently added to my main playlist two new albums, one called **"My Favorite"** and one called **"+01"**. 
When I sorted the playlist with the **"Sort by Imported"** button to see the two new albums in the bottom of the list, however, the album **My Favorite** suddenly split into two. 
![2021-12-17_15-57_1](https://user-images.githubusercontent.com/33501455/146565336-1936405f-9bf1-4e06-a3d4-ebd3a126dafc.png)

On the contrary, sorting by filepath/artist... or any other option doesn't split the album.
![2021-12-17_15-57](https://user-images.githubusercontent.com/33501455/146565328-b128d20a-2e10-4dcb-909f-a758e8e1d5b9.png)

I noticed that this is happening only when the two albums **My Favorite** and **+01** are together. If I delete from the playlist the **+01** album, sorting the playlist **"by Imported"** doesn't split the album **My Favorite**.

Both the two albums are in the same folder and have been tagged with kid3. I'll show proof if needed.

**Comment by Taiko2k** [comment]
The imported order sort and the sort by filepath do use slightly different algorithms. Its just a matter of that being what it is I think. The main issue is the tracks all being in the same folder, where at the import stage Tauon expects that each folder is its own album.

The only odd thing about what you said to me is in regards to having removed the +01 album then doing *sort by imported* is that in that case the tracks should still be showing in an incorrect order?

It would be interesting to see the naming scheme of these files, that might give an idea of where it's tripping up. (At the import stage, the filename is all the importer has to go on, since the tag scan is done later)

**Comment by liu3631** [comment]
> The main issue is the tracks all being in the same folder

I'm sorry, I haven't explained it clearly. What I meant by "in the same folder" is that every track is in its own album folder, and both folders are inside one other folder (artist folder).
![2021-12-17_18-28](https://user-images.githubusercontent.com/33501455/146584263-e2a43e04-5130-43d6-906c-8045af4a7f52.png)

And these are the names of the files:
![2021-12-17_18-30](https://user-images.githubusercontent.com/33501455/146584637-96586bd2-0669-40c9-81f1-942b0b6876f2.png)
![2021-12-17_18-31](https://user-images.githubusercontent.com/33501455/146584640-bcef416b-d9f0-4f31-a40b-07eb1c9d3a86.png)

**Comment by Taiko2k** [comment]
Okay that's really weird then. Are you sure that the album is split just after having done "sort by imported"? (And that there isn't a generator code applied to the playlist)

**Comment by liu3631** [comment]
Yes, I'm sure. The generator code is the default one, I've never touched it. I tried to delete all the tags but that didn't change anything. I'll attach a short video to demonstrate what's happening.

https://user-images.githubusercontent.com/33501455/146586388-11932527-1d39-4174-9a44-c8a7e10f6555.mp4

**Comment by Taiko2k** [comment]
That really shouldn't happen. The only case I can think of in which it might is if you historically imported the album with tracks missing in the folder to which you added at a later point? y/n?

**Comment by liu3631** [comment]
The only thing I can think of is that I've edited the tags of the tracks and albums' folders names. I then made a new folders and tags scan on Tauon.
By experimenting a bit I found out that changing the folder name of the "My Favorite" album from "My Favorite" to something else fixes the problem. When I rescan folders and I "Sort by Imported" the album doesn't split. However, as soon as I rename it back to "My Favorite" and rescan again, the problem comes back.

**Comment by C0rn3j** [comment]
Heya, are you able to reproduce this on the latest [development build](https://github.com/Taiko2k/Tauon/wiki/Development-builds)?

If you still happen to have the files after such a long time that is…

**Comment by Taiko2k** [comment]
Though this may not be a valid bug as what the "Import order" is isn't strictly defined if the tracks are imported at the same time.
